### PR TITLE
Automatically label generated resources with trigger name.

### LINKED ISSUE #169: Proposal to add arbitrary labels for generated resources
# Expected Behavior
@dibbles and I have been pondering how to see what PipelineRuns/Pipelineresources were created by certain triggers (for our use case, so we can monitor them and report on GitHub if the Tasks all passed)

Example labels would be, for example, the Git repository URL, branch, the trigger name, any other params etc

# Actual Behavior
This isn't here atm

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

For any resource created by a trigger, the EventListener will add the
label `tekton.dev/trigger: <trigger name>` as a resource label.

Additional labels can be added by simply setting them as parameters in
the resources themselves.

Fixes #169 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
EventListeners now label generated resources with `tekton.dev/trigger: <trigger name>`.

This improves traceability of generated resources by including the name of the trigger that generated it.
```

### CODE REVIEW

File: pkg/sink/sink.go [hunk 1]
```diff
@@ -167,35 +167,39 @@ func createResources(resources []json.RawMessage, restClient restclient.Interfac
 
 // createResource uses the kubeClient to create the resource defined in the
 // TriggerResourceTemplate and returns any errors with this process
-func createResource(rt json.RawMessage, restClient restclient.Interface, discoveryClient discoveryclient.DiscoveryInterface, eventListenerNamespace string, eventListenerName string, eventID string, logger *zap.SugaredLogger) error {
+func (r Resource) createResource(rt json.RawMessage, triggerName string, eventID string) error {
```
Comment (dibyom) [inline pkg/sink/sink.go]: Nice. Thanks for the cleanup!

File: pkg/sink/sink.go [hunk 2]
```diff
@@ -167,35 +167,39 @@ func createResources(resources []json.RawMessage, restClient restclient.Interfac
...
+		namespace = r.EventListenerNamespace
 	}
-	apiResource, err := findAPIResource(discoveryClient, apiVersion, kind)
+	apiResource, err := findAPIResource(r.DiscoveryClient, apiVersion, kind)
 	if err != nil {
 		return err
 	}
 
-	rt, err = sjson.SetBytes(rt, "metadata.labels."+triggersv1.LabelEscape+triggersv1.EventListenerLabelKey, eventListenerName)
+	rt, err = sjson.SetBytes(rt, "metadata.labels."+triggersv1.LabelEscape+triggersv1.EventListenerLabelKey, r.EventListenerName)
```
Comment (dibyom) [inline pkg/sink/sink.go]: Tiny nit: Might be worth extracting the three calls to add labels to its own helper function for easy testing

Comment (wlynch) [reply pkg/sink/sink.go]: Done!

**Comment by wlynch** [comment]
> This should be really useful for traceability. Thanks @wlynch !
> One minor nit and needs a rebase due to #211 . Otherwise LGTM!

Rebase done.

**Pull request merged**